### PR TITLE
Fix for mempool not clearning

### PR DESCRIPTION
**Pull request by SWvheerden**
## Description

Changes the syncing service to use the local node interface to submit blocks rather than just adding it. 
Submit block now requires you to say if it should propagate the block or not. 
Submit block also returns the result of the add_block to db_backend. 

## Motivation and Context

Currently if the statemachine syncs blocks, the mempool is not given the blocks to clear its pool. The mempool only listens for the addblock event to trigger its "cleansing". But because there is a bug in block propagation, nodes sync rather than get new blocks via propagation. This causes that the node's mempool is never cleared of all transactions. 

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [x] I'm merging against the `development` branch.
* [x] I ran `cargo-fmt --all` before pushing.
* [x] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by neonknight64 — DISMISSED** [review]
LGTM

File: base_layer/core/src/base_node/comms_interface/inbound_handlers.rs [hunk 1]
```diff
@@ -287,7 +290,7 @@ where T: BlockchainBackend + 'static
                 BlockAddResult::OrphanBlock => false,
                 BlockAddResult::ChainReorg(_) => true,
             };
-            if propagate {
+            if propagate && *propagate_out {
```
Comment (neonknight64) [inline base_layer/core/src/base_node/comms_interface/inbound_handlers.rs]: I wonder if the code readability can be improved here.

Comment (SWvheerden) [reply base_layer/core/src/base_node/comms_interface/inbound_handlers.rs]: Perhaps I could rename the first one to Internal_propagate_check

**Review by CjS77 — COMMENTED** [review]
Looks right, but agree with @neonknight64 that it'll be tough knowing what the bool is about for someone looking at the code with no context.

Firstly, there are no comments / docs explaining what it is.
Secondly, readability could be made much better / self documenting by using an Enum or Wrapped struct

`enum BlockOrigin` or `BroadcastBlock(bool)`

**Review by CjS77 — APPROVED** [review]
LGTM

**Pull request merged**